immutablecollections/_immutableset.py

```diff
@@ -61,30 +62,29 @@ def immutableset(
         return iterable
 
     if not disable_order_check:
-        # case where iterable is an ImmutableSet (which is order-safe)
-        # is already checked above
-        if isinstance(iterable, AbstractSet):
-            raise ValueError(
-                "Attempting to initialize an ImmutableSet from "
-                "a non-ImmutableSet set. This probably loses "
-                "determinism in iteration order.  If you don't care "
-                "or are otherwise sure your input has determinstic "
-                "iteration order, specify disable_order_check=True"
-            )
         # we put this check on the outside to avoid isinstance checks on newer Python versions
         if not DICT_ITERATION_IS_DETERMINISTIC:
-            if (
-                isinstance(iterable, KeysView)
-                or isinstance(iterable, ValuesView)
-                or isinstance(iterable, ItemsView)
-            ):
+            if isinstance(iterable, ViewTypes):
```

Comment:
Since the structure of the conditional has changed, is there a reason not to change this to a single if, `if not DICT_ITERATION_IS_DETERMINISTIC and isinstance(iterable, ViewTypes):`?

Reply:
That's what I originally had, but there is performance hit (I think?) from not being able to short-circuit the conditional:
- [combined](https://gist.githubusercontent.com/berquist/eb456fa3d64459d83accb819930d5001/raw/13ca271279c580e6f15b1224ff48b747872d9936/bench_combined.txt)
- [separate](https://gist.githubusercontent.com/berquist/eb456fa3d64459d83accb819930d5001/raw/13ca271279c580e6f15b1224ff48b747872d9936/bench_separate.txt)

Reply:
As made clear by your comment, the multi-if version is clearly faster. Fascinating!